spec/std/channel_spec.cr

```diff
@@ -889,4 +889,31 @@ describe "buffered" do
     ch = Channel(Int32).new(10)
     ch.pretty_inspect.should eq("#<Channel(Int32):0x#{ch.object_id.to_s(16)}>")
   end
+
+  it "can send non blocking" do
+    ch = Channel(Int32).new
+    spawn(same_thread: true) do
+      ch.receive.should eq 1
+    end
+    Fiber.yield # required to let the receiving fiber start
```

Comment:
I'd suggest to use the channel for synchronization between fibers. We already have it anyways and that's more robust than depending on `Fiber.yield` and `same_thread: true` to run the spawned fiber at that point.

```suggestion
    spawn do
      ch.send 0 # sync
      ch.receive.should eq 1
    end
    ch.receive # let the receiving fiber start
```